### DISCUSSION TITLE
Seed user to automatically login

### DISCUSSION THREAD

**Discussion by Putnug1122** [codeigniter4/shield#497] [Q&A] [answered]
How to seed hundred of users to database with login ready ?

**Comment by kenjis** [comment]
You can crerate a user with Fake. Something like this:
https://github.com/codeigniter4/shield/blob/5e6d5175da45b06dbe7d1deda03458d79d45a951/tests/Authentication/Authenticators/SessionAuthenticatorTest.php#L407-L412

**Reply by Putnug1122** [reply]
> You can crerate a user with Fake. Something like this:
> 
> https://github.com/codeigniter4/shield/blob/5e6d5175da45b06dbe7d1deda03458d79d45a951/tests/Authentication/Authenticators/SessionAuthenticatorTest.php#L407-L412

it returns `Call to undefined function App\Controllers\fake()`, it's there any package to import ?

**Reply by MGatner** [reply]
You need to load the framework's Test Helper where `fake()` is defined if you want to use it. 
```php
helper('test');
$user = fake(UserModel::class, [
...
```

**Comment by lonnieezell** [comment, marked as answer]
I believe you have to load the test helper.

**Reply by Putnug1122** [reply]
how to do it exactly 😅, I'm new to shield

**Reply by datamweb** [reply]
Welcome!
```php
helper('test');
/** @var User $user */ 
 $user = fake(UserModel::class, ['username' => 'foorog']); 
 $user->createEmailIdentity([ 
     'email'    => 'FOO@example.com', 
     'password' => 'secret123', 
 ]); 
```
more info [see ](https://codeigniter.com/user_guide/helpers/test_helper.html).

**Reply by Putnug1122** [reply]
it works great! thank you so much
